### ISSUE TITLE
Reproducing results and arguments on LSTM

### ISSUE THREAD

**Issue by tami1082** [google-deepmind/neural_networks_chomsky_hierarchy#9]
Hello there,

First off, big fan of your work with this model—it’s really well done!

I’ve been trying to get it to perform better on three datasets: Odds First (24.76), Duplicate String (31.96), and Modular Arithmetic (31.71). I've played around with various LSTM training parameters but haven't hit the mark yet.

Here are the parameter settings I have used within sweep:
- **Hidden Size**: I have tried 256, 512 and 1024.
- **Learning Rate (lr)**: I experimented with 0.0001, 0.0003, and 0.0005.
- **Seed**: I have tested all values from 0 to 9.

Additionally, I have set both the `modulus` and `vocab_size` parameters to 5. Based on your experience, do you think these settings are appropriate, or how should they be adjusted to better suit the above tasks?

I would greatly appreciate any advice on optimizing these parameters or any documentation and resources you could share that might be helpful.
Thank you for your time and assistance!

Best regards,
Tian

**Comment by fazega** [comment]
Hello,
- 24.76, 31.96 and 31.71 are the average accuracies over lengths [41, 500] you get? They should be >50%, the random baseline.
- Is your question about matching results in the paper, or improving the results of the paper?

**Comment by tami1082** [comment]
1. yes they are the average results on [41, 100].
2. now I would like to match the results.

**Comment by fazega** [comment]
I just launched one training with these parameters (keep the other defaults):
```
seed=0,
model_init_seed=3,
training_steps=10_000,
log_frequency=100,
learning_rate=1e-4,
compute_full_range_test=True,
max_range_test_length=100,
range_test_total_batch_size=512,
range_test_sub_batch_size=64
architecture_params.hidden_size=256
architecture='lstm'
task=OddsFirst(vocab_size=2)
 ```
 
 Results:
 `Network score: 0.5245555384810698`
 Which is what is expected. The train losses and accuracies look perfectly normal too.
 Please check your hardware if you confirm you are using exactly the same parameters.

**Comment by fazega** [comment]
Possible problem: are you sure you are using a vocabulary size of 2 in Odds First and Duplicate String?

**Comment by tami1082** [comment]
🤔I use 5 in both datasets

**Comment by fazega** [comment]
Yes ok, so all your results are expected (baseline is random at 20%) 👍
The paper uses a vocab size of 2 in these two tasks.

**Comment by tami1082** [comment]
ok!! Thank you!!!! 
🥳

**Comment by fazega** [comment]
I agree this is not very explicit in the paper (we just say the baseline is at 50% for these tasks in table 2). I'll update the code to make it clear there at least.
And 32% on Duplicate String is quite good for a baseline of 20% actually: length generalization is hard!
